src/test/groovy/graphql/parser/ParserTest.groovy

```diff
@@ -964,6 +961,22 @@ triple3 : """edge cases \\""" "" " \\"" \\" edge cases"""
         '1.2e3e'  | _
     }
 
+    @Unroll
+    def 'parse ast field definition #valueLiteral'() {
+        expect:
+        def fieldDefinition = Parser.parseFieldDefinition(valueLiteral)
+        AstPrinter.printAstCompact(fieldDefinition) == valueLiteral
+
+        where:
+        valueLiteral                                              | _
+        'foo: Foo'                                                | _
+        'foo(a:String): Foo'                                      | _
+        'foo(a:String!,b:Int!): Foo'                              | _
+        'foo(a:String! ="defaultValue",b:Int!): Foo'              | _
+        'foo(a:String!,b:Int!): Foo @directive(someValue:String)' | _
+    }
```

Comment:
What about the failure case - what happens when it fails to parse?  Will it throw and exception and if so lets lock that in with a test